datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -181,27 +185,37 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
 
                 auth = requests_ntlm.HttpNtlmAuth(config['ntlm_domain'], config['password'])
 
-        if auth is None and config['kerberos_auth']:
-            ensure_kerberos()
+        if auth is None:
+            if config['kerberos_auth']:
+                ensure_kerberos()
 
-            # For convenience
-            if is_affirmative(config['kerberos_auth']):
-                config['kerberos_auth'] = 'required'
+                # For convenience
+                if is_affirmative(config['kerberos_auth']):
+                    config['kerberos_auth'] = 'required'
 
-            if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
-                raise ConfigurationError(
-                    'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
-                        config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
+                if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
+                    raise ConfigurationError(
+                        'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
+                            config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
+                        )
                     )
+
+                auth = requests_kerberos.HTTPKerberosAuth(
+                    mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
+                    delegate=is_affirmative(config['kerberos_delegate']),
+                    force_preemptive=is_affirmative(config['kerberos_force_initiate']),
+                    hostname_override=config['kerberos_hostname'],
+                    principal=config['kerberos_principal'],
                 )
+            elif config['aws_host']:
+                ensure_aws()
+
+                if not config['aws_region']:
+                    raise ConfigurationError('AWS auth requires the setting `aws_region`')
 
-            auth = requests_kerberos.HTTPKerberosAuth(
-                mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
-                delegate=is_affirmative(config['kerberos_delegate']),
-                force_preemptive=is_affirmative(config['kerberos_force_initiate']),
-                hostname_override=config['kerberos_hostname'],
-                principal=config['kerberos_principal'],
-            )
+                auth = requests_aws.BotoAWSRequestsAuth(
```

Comment:
Note: That means we're not able to make requests to multiple AWS services or regions from the same instance. Do we expect to need this in the future?

Reply:
The same could be said for Kerberos auth, I think if we encounter that we can easily handle it in our integrations

Reply:
(I'm marking the comment as "unsolved" since it's not really solved and could be useful for readers)